packages/io_library/lib/src/io_handler.dart

```diff
@@ -201,8 +203,19 @@ class IOHandler {
     final canvasState = ref.read(canvasStateProvider);
     final imgWidth = canvasState.size.width.toInt();
     final imgHeight = canvasState.size.height.toInt();
-    final catrobatImage = CatrobatImage(
-        commands, imgWidth, imgHeight, canvasState.backgroundImage);
+    Uint8List? backgroundImageData;
+    if (canvasState.backgroundImage != null) {
+      final result = await ref
+          .read(IImageService.provider)
+          .exportAsPng(canvasState.backgroundImage!);
```

Comment:
Don't use !
Do:
final backgroundImage = canvasState.backgroundImage;
if (backgroundImage != null) 

Then dart knows that this cannot be null